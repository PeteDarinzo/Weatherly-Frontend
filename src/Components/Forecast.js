import React from "react";
import ForecastCard from "./ForecastCard";
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';
import Box from '@mui/material/Box';
import { v4 as uuid } from "uuid";
import Grid from '@mui/material/Grid';
import ForecastPaper from "./ForecastPaper";
import { useDispatch, useSelector } from "react-redux";
import { fetchForecastFromAPI } from "../Actions/actions";
import { Typography } from "@mui/material";


const Forecast = () => {

  const userData = useSelector(store => store.user);
  const forecast = useSelector(store => store.forecast);
  const dispatch = useDispatch();

  const { lat, lon, city, minTemp, maxTemp, units, conditions } = userData;

  function handleClick() {
    dispatch(fetchForecastFromAPI(lat, lon, units));
  }

  return (
    <Container>
      <Box m={4}>
        <Button onClick={handleClick} variant="contained">Refresh</Button>
      </Box>
      <Typography variant="h3">Forecast for {city}</Typography>
      <Grid container >
        {forecast.map(day => {
          return (
            <Grid item xs={12} sm={6} md={3} key={uuid()}>

              <ForecastPaper day={day.name} min={day.temp.min} max={day.temp.max} feelsLike={day.feels_like.day} description={(day.weather[0]).description} icon={(day.weather[0]).icon} />

              {/* <ForecastCard min={day.temp.min} max={day.temp.max} feelsLike={day.feels_like.day} description={(day.weather[0]).description} icon={(day.weather[0]).icon} /> */}
            </Grid>
          )
        })}
      </Grid>
    </Container>
  );
}

export default Forecast;