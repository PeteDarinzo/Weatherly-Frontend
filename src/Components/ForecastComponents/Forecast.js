import React, { useState, useEffect } from "react";
import ForecastCard from "./ForecastCard";
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';
import Box from '@mui/material/Box';
import { v4 as uuid } from "uuid";
import Grid from '@mui/material/Grid';
import ForecastPaper from "./ForecastPaper";
import { useDispatch, useSelector } from "react-redux";
import { fetchForecastFromAPI } from "../../Actions/actions";
import { Typography } from "@mui/material";


const Forecast = () => {

  const userData = useSelector(store => store.user);
  const { lat, lon, city, minTemp, maxTemp, units, thunderstorm, drizzle, rain, snow, overcast } = userData;
  const forecast = useSelector(store => store.forecast);
  const dispatch = useDispatch();
  const [userPrefs, setUserPrefs] = useState({})

  useEffect(() => {

    if (userData) {
      setUserPrefs({minTemp, maxTemp, thunderstorm, drizzle, rain, snow, overcast });
    }

  }, [userData]);


  function handleClick() {
    console.log(userPrefs);
    dispatch(fetchForecastFromAPI(lat, lon, units));
  }

  function getCompatibility(day) {
    let conditionsMet;
    let tempMet;
    const conditionCode = day.weather[0].id;
    const temperature = day.feels_like.day;
    if (((200 <= conditionCode) && (conditionCode <= 232)) && (userPrefs.thunderstorm)) {
      conditionsMet = true;
    } else if (((300 <= conditionCode) && (conditionCode <= 321)) && (userPrefs.drizzle)) {
      conditionsMet = true;
    } else if (((500 <= conditionCode) && (conditionCode <= 531)) && (userPrefs.drizzle)) {
      conditionsMet = true;
    } else if (((600 <= conditionCode) && (conditionCode <= 622)) && (userPrefs.snow)) {
      conditionsMet = true;
    } else if (((801 <= conditionCode) && (conditionCode <= 804)) && (userPrefs.overcast)) {
      conditionsMet = true;
    } else {
      conditionsMet = false
    }



    if(temperature <= userPrefs.minTemp || userPrefs.maxTemp <= temperature) {
      tempMet = true;
    }

    if(conditionsMet && tempMet) {
      return "Very Good!";
    } else if (conditionsMet || tempMet) {
      return "Good!";
    } else {
      return "Neutral!";
    }
  }

  return (
    <Container>
      <Box m={4}>
        <Button onClick={handleClick} variant="contained">Refresh</Button>
      </Box>
      <Typography variant="h3">Forecast for {city}</Typography>
      <Grid container spacing={3}>
        {forecast.map(day => {

          let compatibility = getCompatibility(day);

          return (
            <Grid item xs={12} sm={6} md={3} key={uuid()}>

              <ForecastPaper day={day.name} min={day.temp.min} max={day.temp.max} feelsLike={day.feels_like.day} description={(day.weather[0]).description} icon={(day.weather[0]).icon} units={units === "imperial" ? "F" : "C"} compatibility={compatibility} />

              {/* <ForecastCard min={day.temp.min} max={day.temp.max} feelsLike={day.feels_like.day} description={(day.weather[0]).description} icon={(day.weather[0]).icon} /> */}
            </Grid>
          )
        })}
      </Grid>
    </Container>
  );
}

export default Forecast;