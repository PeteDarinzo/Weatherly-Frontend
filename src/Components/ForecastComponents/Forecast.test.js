import Forecast from "./Forecast";
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter, Route } from "react-router-dom";

it("renders forecast without crashing", () => {
  render(
    <Forecast />
  );
});