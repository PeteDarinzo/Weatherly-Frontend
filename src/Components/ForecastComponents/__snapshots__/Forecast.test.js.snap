// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the forecast snapshot 1`] = `
<DocumentFragment>
  <b>
    Loading forecast
  </b>
</DocumentFragment>
`;
